mycore-solr/src/main/java/org/mycore/solr/MCRSolrCore.java

```diff
@@ -167,12 +169,32 @@ private static HttpSolrClientBase getSolrClientInstance(String baseSolrUrl, int
                 baseBuilder.useHttp1_1(true);
             });
 
-        return baseBuilder
+        HttpSolrClientBase client = baseBuilder
             .withConnectionTimeout(connectionTimeout, TimeUnit.MILLISECONDS)
             .withIdleTimeout(socketTimeout, TimeUnit.MILLISECONDS)
             .withRequestTimeout(socketTimeout, TimeUnit.MILLISECONDS)
             .withRequestWriter(new BinaryRequestWriter())
             .build();
+
+        // MCR-3445 use Java Reflection to increase the size of the RequestBuffer of the internal HTTP client.
+        // This is necessary to send word coordinates for alto files via MCRSolrFileIndexHandler.
+        if (client instanceof Http2SolrClient solrclient) {
+            try {
+                Method mGetHttpClient = solrclient.getClass().getDeclaredMethod("getHttpClient");
+                mGetHttpClient.setAccessible(true);
+
+                // https://github.com/jetty/jetty.project/blob/jetty-11.0.x/jetty-client/src/main/java/org/eclipse/jetty/client/HttpClient.java
+                Object jettyHttpClient = mGetHttpClient.invoke(solrclient);
+                
+                Method mSetRequestBufferSize = jettyHttpClient.getClass()
+                    .getDeclaredMethod("setRequestBufferSize", int.class);
+                mSetRequestBufferSize.invoke(jettyHttpClient, 511 * 1024);
```

Comment:
Why this specific hard-coded value? Could be a configuration property for applications that need an even larger buffer size.